### PR TITLE
[csrng] Select Canright S-Box implementation for AES cipher core

### PR DESCRIPTION
**Pull request by vogelpi**
While checking top_earlgrey for ways to save FPGA logic utilization, I noted that CSRNG always uses the LUT-based S-Box for the embedded AES cipher core. This is good for FPGAs but for ASICs, the unmasked Canright S-Box should be used.

Therefore, this PR changes the default selection for the S-Box inside the embedded AES cipher core from the LUT-based implementation to the unmasked Canright design. This should reduce the area footprint of CSRNG a little bit. The LUT-based implementation can explicitly be selected when working with FPGA targets.

### CODE REVIEW

**Review by mwbranstad — APPROVED** [review]
I see making Canright the default, and only exception is the FPGA. LGTM

**Comment by vogelpi** [comment]
Thanks @mwbranstad for having a look and approving this PR. Once I get approval from a committer, I will merge this.

**Review by imphil — APPROVED** [review]

**Comment by vogelpi** [comment]
Thanks @imphil :+1:

**Pull request merged**
